File: Java/Ficheros/Practica1.java
/**
 * Escribe un programa que guarde en un fichero con nombre primos.dat los
 * números primos que hay entre 1 y 500.
 * 
 * @author Cristina Prieto Jalao
 *
 */

import java.io.BufferedWriter;
import java.io.FileWriter;
import java.io.IOException;

public class Practica1 {
  public static void main (String [] args) throws IOException {
    BufferedWriter bw = new BufferedWriter(new FileWriter("primos.dat"));
      for (int i = 1; i < 501; i++) {
        if (esPrimo(i)) {
          bw.write((i) + "\n");
        }
      }
      bw.close();
  }
    
    /**
     * Método para saber si un número es primo o no
     * @param x
     * @return true si es primo. false si no es primo
     */
    public static boolean esPrimo(int x) {
      for (int i = 2; i < x; i++) {
        if ((x % i) == 0) {
          return false;
        }
      }
      return true;
  }

}

File: Python/Ficheros/Practica1.py
'''
Escribe un programa que guarde en un fichero con nombre primos.dat los
números primos que hay entre 1 y 500.

@author: Cristina Prieto Jalao
'''
archivo = open('datos.dat','w')

def esPrimo(num):
  if num<2:
    return False
  for i in range (2,num):
    if num%i==0:
      return False
    return True
  
numPrimos=[]

for i in range(1, 500):
  if esPrimo(i)==True:
    numPrimos.append(i)
    
print(numPrimos)

for i in numPrimos:
    archivo.write(str((i))+'\n')
    
archivo.close()